File: backend/app/schemas/product_variant.py
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field


class ProductVariantCreate(BaseModel):
    product_id: int
    color: str = Field(min_length=2, max_length=50)
    size: str = Field(min_length=1, max_length=20)
    cost_price: Decimal = Field(ge=0, decimal_places=2)
    sale_price: Decimal = Field(ge=0, decimal_places=2)
    minimum_stock: int = Field(default=0, ge=0)


class ProductVariantResponse(BaseModel):
    id: int
    company_id: int
    product_id: int
    sku: str
    color: str
    size: str
    cost_price: Decimal
    sale_price: Decimal
    minimum_stock: int
    active: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/productVariant.ts
export interface ProductVariant {
  id: number
  company_id: number
  product_id: number
  sku: string
  color: string
  size: string
  cost_price: string
  sale_price: string
  minimum_stock: number
  active: boolean
  created_at: string
  updated_at: string
}

export interface ProductVariantCreate {
  product_id: number
  color: string
  size: string
  cost_price: number
  sale_price: number
  minimum_stock: number
}